utils/validate-quickstart-data-source.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  fetchPaginatedGHResults,
+  filterQuickstartConfigFiles,
+  filterOutTestFiles,
+  GithubAPIPullRequestFile,
+  isNotRemoved,
+} from './lib/github-api-helpers';
+
+import Quickstart from './lib/Quickstart';
+import DataSource from './lib/DataSource';
+import { passedProcessArguments } from './lib/helpers';
+
+export const validateDataSourceIds = (
+  githubFiles: GithubAPIPullRequestFile[]
+) => {
+  const quickstartsWithInvalidDataSources = filterQuickstartConfigFiles(
+    githubFiles
+  )
+    .filter(isNotRemoved)
+    .map(({ filename }) => new Quickstart(filename))
+    .map((quickstart)=> {
+      const invalidIds = quickstart.config.dataSourceIds?.filter((dataSourceId)=> {
+        return !new DataSource(dataSourceId).isValid;
```

Comment:
If I attempt to use a core data source on a quickstart, this will fail. Could we possibly pull in the list of data sources to check against from Nerdgraph?
The only other option I see is letting this fail for core data sources and getting the success from the validation against Nerdgraph. However, I can't think of a way to say that one or the other github actions must succeed to merge.